MAUI/DataGrid/column-drag-and-drop.md

```diff
@@ -0,0 +1,226 @@
+
+Dropping when dragging a particular column can be canceled using the `DataGridDragAction` argument of the `QueryColumnDragging` event handler.
+
+{% highlight c# %}
+dataGrid.QueryColumnDragging += SfDataGrid_QueryColumnDragging;
+
+private void SfDataGrid_QueryColumnDragging(object? sender, DataGridQueryColumnDraggingEventArgs e)
+{
+    //e.To returns the index of the current column.
+    //e.DraggingAction returns the dragging status of the column.
+    if ((e.To > 3 && e.To < 6) &&
+    (e.DraggingAction == DataGridDragAction.DragEnded || e.DraggingAction == DataGridDragAction.Dragging))
+    {
+        e.Cancel = true;
+    }
+}
+{% endhighlight %}
+
+## Cancel dropping for a particular column
+
+The dropping of a particular column can be canceled using the `DataGridDragAction` argument of the `QueryColumnDragging` event handler.
+
+{% highlight c# %}
+dataGrid.QueryColumnDragging += SfDataGrid_QueryColumnDragging;
+
+private void SfDataGrid_QueryColumnDragging(object? sender, DataGridQueryColumnDraggingEventArgs e)
+{
+    //e.From returns the index of the dragged column.
+    //e.DraggingAction returns the dragging status of the column.
+    if (e.From == 1 && e.DraggingAction == DataGridDragAction.DragEnded)
+        e.Cancel = true;
+}
+{% endhighlight %}
+
+## Cancel dropping at a particular position
+
+Dropping at a particular position can be canceled using the `DataGridDragAction` argument of the `QueryColumnDragging` event handler.
+
+{% highlight c# %}
+dataGrid.QueryColumnDragging += SfDataGrid_QueryColumnDragging;
+
+private void SfDataGrid_QueryColumnDragging(object? sender, DataGridQueryColumnDraggingEventArgs e)
+{
+    //e.From returns the index of the dragged column.
+    //e.DraggingAction returns the dragging status of the column.
+    if ((e.To == 3 || e.To == 5) && e.DraggingAction == DataGridDragAction.DragEnded)
+        e.Cancel = true;
+}
+{% endhighlight %}
+
+## Cancel dropping of a particular column in a position
+
+Dropping of a particular column in a position can be canceled using the `DataGridDragAction` and `DraggingPosition` arguments of the `QueryColumnDragging` event handler.
+
+{% highlight c# %}
+dataGrid.QueryColumnDragging += SfDataGrid_QueryColumnDragging;
+
+private void SfDataGrid_QueryColumnDragging(object? sender, DataGridQueryColumnDraggingEventArgs e)
+{
+    //e.From returns the index of the dragged column.
+    //e.DraggingPosition returns the x and y position of the current column
+    if (e.DraggingPosition == new Point(100, 100) && e.DraggingAction == DataGridDragAction.DragEnded)
+        e.Cancel = true;
+}
+{% endhighlight %}
+
+## Cancel drag and drop between frozen and non-frozen columns
+
+### Cancel dragging between frozen and non-frozen columns
+
+Dragging between frozen and non-frozen columns can be canceled using the `DataGridDragAction` and `From` arguments of the `QueryColumnDragging` event handler, by checking whether the value of the `From` argument is a frozen column index.
```

Comment:
The dragging between frozen and non-frozen columns can be canceled using the `DataGridDragAction` and `From` arguments of the `QueryColumnDragging` event, by checking whether the value of the `From` argument is a frozen column index.

Reply:
Modified